cmd/krel/cmd/release_notes.go

```diff
@@ -113,12 +141,85 @@ func runReleaseNotes() (err error) {
 	logrus.Infof("Using start tag %v", start)
 	logrus.Infof("Using end tag %v", tag)
 
-	_, err = releaseNotesFrom(start)
+	if releaseNotesOpts.sigReleaseFork == "" && releaseNotesOpts.jsonFile == "" && releaseNotesOpts.mdFile == "" {
+		logrus.Fatal("No output selected. You have to specify at least one of --md-file --json-file or --sigrelease-fork")
+	}
+
+	// TODO: Check output flags, dirs and repos before running the gatherer
+
+	result, err := releaseNotesFrom(start)
 	if err != nil {
 		return errors.Wrapf(err, "generating release notes")
 	}
 
-	//TODO: implement PR creation for k-sigs/release-notes and k/sig-release
+	// Create RN draft PR
+	if releaseNotesOpts.sigReleaseFork != "" {
+		err = createDraftPR(tag, result)
+		if err != nil {
+			return errors.Wrap(err, "Failed to create release notes draft PR")
+		}
+	}
+
+	if releaseNotesOpts.mdFile != "" {
+		err = ioutil.WriteFile(releaseNotesOpts.mdFile, []byte(result.markdown), 0644)
+		if err != nil {
+			return errors.Wrap(err, "Failed to write markdown file")
+		}
+	}
+
+	if releaseNotesOpts.jsonFile != "" {
+		err = ioutil.WriteFile(releaseNotesOpts.jsonFile, []byte(result.json), 0644)
+		if err != nil {
+			return errors.Wrap(err, "Failed to write JSON file")
+		}
+	}
+
+	// TODO: implement PR creation for k-sigs/release-notes
+	return nil
+}
+
+// createDraftPR pushes the release notes draft to the users fork
+func createDraftPR(tag string, result *releaseNotesResult) error {
+	repoparts := strings.Split(releaseNotesOpts.sigReleaseFork, "/")
```

Comment:
Actually, we should probably add the `validatePROptions()` check here instead.
`repoparts` becomes unnecessary if the flags get split as suggested above.

Reply:
I chose to call `validatePROptions()` before generating the notes. i wanted to avoid running the release notes gatherer (which is very expensive timewise) if the PR was going to fail anyway.